### PR TITLE
Require `toolz` & soften `cytoolz` dependency

### PR DESCRIPTION
**Pull request by kenodegard**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Xref: https://github.com/conda/conda/issues/11698, https://github.com/conda-forge/conda-feedstock/pull/170

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -39,6 +39,7 @@ requirements:
   host:
     - python
     - conda-package-handling >=1.3.0
+    - cytoolz >=0.8.1
```
Comment (jakirkham) [inline recipe/meta.yaml:42]: Do we need `cytoolz` to build `conda`? Or is it just needed at runtime?

Comment (kenodegard) [reply recipe/meta.yaml:42]: During the build we run `conda init --install`

Comment (jakirkham) [reply recipe/meta.yaml:42]: Ah I see and so that imports `cytoolz`?

Comment (kenodegard) [reply recipe/meta.yaml:42]: https://github.com/conda-forge/conda-feedstock/blob/78cc35ef3614df48de1fab2e3eeb83b0c85fd565/recipe/build.sh#L19

Comment (kenodegard) [reply recipe/meta.yaml:42]: yeah see the failing canary builds: https://github.com/conda/conda/runs/7731850991?check_suite_focus=true#step:3:761

Comment (kenodegard) [reply recipe/meta.yaml:42]: This isn't required for 4.13 or 4.14, but will be necessary for releases after that

**Pull request merged**